sony.go

```diff
@@ -0,0 +1,221 @@
+// http.StatusText().
+func (err HTTPStatusError) Error() string {
+	return http.StatusText(int(err))
+}
+
+// Unwrap returns ErrHTTP for all errors so HTTPStatusErrors can be checked
+// with `errors.Is(err, ErrHTTP)`, instead of the two-line
+// `var errHTTP HTTPStatusErr; errorsAs(err, &errHTTP)`. The latter is still
+// possible if the error type is required.
+func (err HTTPStatusError) Unwrap() error {
+	return ErrHTTP
+}
+
+// SonyError captures an error returned by the Sony REST IP control protocol
+// as an error returned in the payload of an HTTP response. These errors are
+// returned as an error code and a string describing it.
+type SonyError struct {
+	Code    int
+	Message string
+}
+
+// NewSonyError returns a SonyError from the response. The []any parameter
+// is expected to have two elements, an int (code) and a string (message).
+// If the size or types are not as just described, ErrInvalidSonyResponse
+// is returned.
+func NewSonyError(resp []any) error {
+	if len(resp) != 2 {
+		return ErrInvalidSonyResponse
+	}
+	code, ok := resp[0].(int)
+	if !ok {
+		return ErrInvalidSonyResponse
+	}
+	msg, ok := resp[1].(string)
+	if !ok {
+		return ErrInvalidSonyResponse
+	}
+	return SonyError{Code: code, Message: msg}
+}
+
+// Error returns the message portion of the error.
+func (err SonyError) Error() string {
+	return err.Message
+}
+
+// Unwrap returns ErrSony for all errors so SonyErrors can easily be checked
+// with `errors.Is(err, ErrSony)` for when you don't care about the specific
+// code or message of the error (when you would use `errors.As()`).
+func (err SonyError) Unwrap() error {
+	return ErrSony
+}
+
+// NewRESTClient creates and returns a BraviaClient reachable at the given
+// hostname, using the Pre-Shared Key given as psk as the password. If psk is
+// the empty string, it is not used.
+func NewRESTClient(hostname, psk string) *RESTClient {
+	return &RESTClient{
+		BaseURL:    "http://" + hostname + "/sony",
+		PSK:        psk,
+		HTTPClient: &http.Client{},
+	}
+}
+
+// none is a type to be used with `post[T]()` for when a response is not returned.
+// e.g. `_, err := post[none](...)`.
+type none struct{}
+
+// post[T] executes a REST IP control command returning the result of type T or
+// an error if the command did not succeed. If no data was returned from the
+// HTTP call, the returned value will be nil. The `none` type can be used when
+// no response is expected:
+//
+//	_, err := post[none](client, service, method, version, params...)
+//
+// The protocol docs define service, method and version. Params are any values
+// that can be marshaled as JSON and are passed in the `params` part of the
+// request.
+func post[T any](bc *RESTClient, service, method, version string, params ...any) (*T, error) {
+	// Timeout after 10s. Arguably that's too long. This doesn't really need to be configurable.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
wouldn't you set this on `bc.HTTPClient.Timeout` ?

Reply:
Done in c4203a825d6445fe2da4ff8efd6320292e2981fe